lightning/src/ln/channel.rs

```diff
@@ -5417,7 +5417,10 @@ impl<SP: Deref> Channel<SP> where
 	}
 
 	fn get_last_revoke_and_ack(&self) -> msgs::RevokeAndACK {
-		let next_per_commitment_point = self.context.holder_signer.as_ref().get_per_commitment_point(self.context.holder_commitment_point.transaction_number(), &self.context.secp_ctx);
+		debug_assert!(self.context.holder_commitment_point.transaction_number() <= INITIAL_COMMITMENT_NUMBER + 2);
+		// TODO: handle non-available case when get_per_commitment_point becomes async
```

Comment:
For my understanding, is this referring to when we implement [this](https://github.com/lightningdevkit/rust-lightning/pull/3086/files#diff-8d28e96b6f27edb9a9739b2dce334f8906ec906155d421a7b5b02aa8e6c96057R1127) TODO to add the new `HolderCommitmentPoint` variant (which I think means that `current_point()` will become `Option`al)?

Reply:
yep! technically it should never be unavailable by the time we get here, but figured i'd leave a note to think about it again when things change

Reply:
It seems like `PendingNext` would also work here, so the assertion below still doesn't totally make sense to me, but it probably gets clearer in the follow-ups.